src/targets/gpu/fuse_mlir.cpp

```diff
@@ -61,12 +62,55 @@ bool mlir_enabled()
 #endif
 }
 
-static bool is_requested(std::string_view option, bool fallback = false)
+namespace {
+struct requested
+{
+};
+struct rejected
+{
+};
+} // namespace
+
+static bool is_negated_op(const std::string& s)
+{
+    if(s.empty())
+        return false;
+    return contains({'!', '~'}, s[0]);
```

Comment:
you don't necessarily need to use `contains` `s[0] == '!'` or s[0] == `~` should suffice.